File: frontend/src/types/rag.ts
import { ClauseCitation } from './policy';

export interface GroundedAnswerResponse {
  query: string;
  answer: string;
  confidence: number;
  citations: ClauseCitation[];
  grounded_clauses: string[];
  suggested_actions: string[];
  policy_name: string;
  language?: string;
}

export interface ChatMessage {
  id: string;
  sender: 'user' | 'assistant';
  text: string;
  timestamp: string;
  citations?: ClauseCitation[];
  suggested_actions?: string[];
}

File: backend/app/schemas/rag.py
from typing import List, Optional
from pydantic import BaseModel
from .policy import ClauseCitation

class RAGQueryRequest(BaseModel):
    policy_id: str
    query: str
    hospital_name: Optional[str] = None
    procedure_name: Optional[str] = None

class GroundedAnswerResponse(BaseModel):
    query: str
    answer: str
    confidence: float
    citations: List[ClauseCitation]
    grounded_clauses: List[str]
    suggested_actions: List[str]
    policy_name: str
